### ISSUE TITLE
bin folder symlinked

### ISSUE THREAD

**Issue by zealot128** [dresden-weekly/ansible-rails#26]
When I run rails console (or other rails commands) from the command line, I get this lengthy message

```
$ bundle exec rails c
Looks like your app's ./bin/rails is a stub that was generated by Bundler.

In Rails 4, your app's bin/ directory contains executables that are versioned
like any other source code, rather than stubs that are generated on demand.

Here's how to upgrade:

  bundle config --delete bin    # Turn off Bundler's stub generator
  rake rails:update:bin         # Use the new Rails 4 executables
  git add bin                   # Add bin/ to source control

You may need to remove bin/ from your .gitignore as well.

When you install a gem whose executable you want to use in your app,
generate it and add it to source control:

  bundle binstubs some-gem-name
  git add bin/new-executable

Loading production environment (Rails 4.2.4)
2.2.0 :001 >
```

In addition, when quitting that irb session, Rails will exit with a lengthy usage message.

When I run the given commands, Rails c does not start anymore, as the symlinking of the `bin` folder makes it confused:

```
$ bundle exec rails c
bin/rails:3:in `require_relative': cannot load such file -- /home/user/app/shared/config/boot (LoadError)
        from bin/rails:3
```

So it seems, either
- the Rails binstubs don't work with symlinked bin folder, or
- the Bundle binstubs generating that huge warnings.

What is the reasoning behind symlinking bin by default? Somebody else experience that problem?

Overriding [`rails_deploy_shared_folders`](https://github.com/dresden-weekly/ansible-rails/blob/develop/rails/create-release/defaults/main.yml#L31) and making sure that `bin/*` is generated by rails fixes that issue for me.

FWIW: Capistrano-bundler decided to remove that bin dir by default: https://github.com/capistrano/capistrano/pull/1321

**Comment by arBmind** [comment]
Thanks for your throughout investigation into this issue.
So far, we did not run into this issue. But we simply might never tried to run such a command on the server.

The ´bin/´ folder and the stubs are only useful to run rails commands without the bundler exec prefix.
These stubs are version agnostic. The should find the right bundler version - if nothing goes wrong.

I will have to take a deeper look into this. (May take a bit of time)

**Comment by zealot128** [comment]
Rails seems to always look for the binstubs and use them, if you run vanilla "bundle exec rails". I've never used binstubs directly.

It works for me, but I have to always modify the default rails_deploy_shared_folders and remove bin from it + remember to add it to exported files/folders from git.

We might also just add a little documentation on it :)

**Comment by arBmind** [comment]
Should be fixed with #39
